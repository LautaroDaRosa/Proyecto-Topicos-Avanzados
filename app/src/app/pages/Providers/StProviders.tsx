import styled from 'styled-components';

const StProviders = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  flex: 1;
  gap: 64px;
  width: 100%;
`;

export default StProviders;
