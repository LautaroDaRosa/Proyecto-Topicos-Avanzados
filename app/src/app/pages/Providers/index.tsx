import React, { useState } from 'react';
import StProviders from './StProviders';
import FilterBar from 'app/components/FilterBar';
import ProvidersGrid from 'app/components/ProvidersGrid';
/* import { useDispatch, useSelector } from 'react-redux';
import { getProvidersSelector } from 'store/providers/selectors';
import { ProvidersSlice } from 'store/providers/slice'; */
import { Provider } from 'store/providers/types';

const Providers = () => {
  const mockedProviders: Provider[] = [
    {
      name: 'Canal 10',
      businessName: 'CANAL 10 S.A.',
      rut: '734879753738964',
      logo: 'https://upload.wikimedia.org/wikipedia/commons/thumb/c/c8/Canal_10_Uruguay_Logo_1960.webp/640px-Canal_10_Uruguay_Logo_1960.webp.png',
      address: 'Lorenzo Carnelli 1234',
      id: 1,
      score: 9,
      categories: ['Comunicaciones, Publicidad'],
    },
    {
      name: 'Canal 4',
      businessName: 'CANAL 4 S.A.',
      rut: '633334234363643',
      logo: 'https://upload.wikimedia.org/wikipedia/commons/3/33/Canal4_uy.png',
      address: 'Paraguay, 2253, Montevideo, Uruguay',
      id: 2,
      score: 6,
      categories: ['Comunicaciones, Publicidad'],
    },
    {
      name: 'Teledoce',
      businessName: 'TELEDOCE S.A.',
      rut: '3457385937529',
      logo: 'https://upload.wikimedia.org/wikipedia/commons/7/79/Teledoce_2022.png',
      address: 'Enriqueta Compte y Riqué 1276',
      id: 3,
      score: 7,
      categories: ['Comunicaciones, Publicidad'],
    },
    {
      name: 'Banco Santander del Uruguay',
      businessName: 'SANTANDER S.A.',
      rut: '5765384337675',
      logo: 'https://upload.wikimedia.org/wikipedia/commons/thumb/6/60/Banco_santander_logo.svg/2560px-Banco_santander_logo.svg.png',
      address: '18 de Julio 1321',
      id: 4,
      score: 9,
      categories: ['Servicios Financieros y Afines'],
    },
  ];
  /* const allProviders: Provider[] = useSelector(getProvidersSelector); */
  const [filteredProviders, setFilteredProviders] = useState(mockedProviders);
  /* const { actions } = ProvidersSlice();
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(actions.fetchProviders());
  }, [dispatch, actions]); */

  console.log(mockedProviders);

  const handleSearch = (
    name: string,
    businessName: string,
    rut: string,
    score: string,
    category: string,
  ) => {
    const filterFn = p => {
      return (
        (name === '' || p.name.toLowerCase().includes(name.toLowerCase())) &&
        (businessName === '' ||
          p.businessName.toLowerCase().includes(businessName.toLowerCase())) &&
        (rut === '' || p.rut === rut) &&
        (score === '' || p.score === parseInt(score)) &&
        (category === '' || p.categories.includes(category))
      );
    };

    setFilteredProviders(mockedProviders.filter(filterFn));
  };

  const handleClearFilters = () => {
    setFilteredProviders(mockedProviders);
  };
  return (
    <StProviders>
      <FilterBar onSearch={handleSearch} clearFilters={handleClearFilters} />
      {filteredProviders.length ? (
        <ProvidersGrid providers={filteredProviders} />
      ) : (
        <span>
          No se han encontrado proveedores para los filtros de búsqueda
          aplicados.
        </span>
      )}
    </StProviders>
  );
};

export default Providers;
