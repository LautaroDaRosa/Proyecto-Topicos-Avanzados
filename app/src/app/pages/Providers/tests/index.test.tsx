import * as React from 'react';
import { render } from '@testing-library/react';

import Providers from '..';

describe('Providers', () => {
  it('should match snapshot', () => {
    const providers = render(<Providers categories={['Cat1']} />);
    expect(providers).toMatchSnapshot();
  });
});
