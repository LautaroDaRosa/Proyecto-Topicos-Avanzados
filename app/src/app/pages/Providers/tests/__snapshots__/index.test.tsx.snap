// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Providers should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  gap: 64px;
  width: 100%;
}

.c2 {
  border-radius: 8px;
  height: 48px;
  font-size: 16px;
  padding: 12px 24px;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  border: 1.5px solid #04b0f4;
  background-color: white;
  color: #04b0f4;
}

.c2:disabled {
  border-color: #c0bfcb;
  color: #c0bfcb;
}

.c2:hover:enabled {
  cursor: pointer;
  color: white;
  background-color: #04b0f4;
}

.c2:active:enabled {
  color: white;
  background-color: #137ca5;
  border-color: #137ca5;
}

.c3 {
  border-radius: 8px;
  height: 48px;
  font-size: 16px;
  padding: 12px 24px;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  border: 1.5px solid #fca408;
  background-color: white;
  color: #fca408;
}

.c3:disabled {
  border-color: #c0bfcb;
  color: #c0bfcb;
}

.c3:hover:enabled {
  cursor: pointer;
  color: white;
  background-color: #fca408;
}

.c3:active:enabled {
  color: white;
  background-color: #b27508;
  border-color: #b27508;
}

.c1 {
  background: white;
  border-radius: 8px;
  padding: 24px;
  box-shadow: 0px 0px 20px 0px lightgrey;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 input {
  height: 40px;
  border-radius: 8px;
  border: 1px solid lightgray;
  padding-left: 12px;
  padding-right: 12px;
}

.c1 select {
  height: 40px;
  border-radius: 8px;
  border: 1px solid lightgray;
  padding-left: 12px;
  padding-right: 12px;
  cursor: pointer;
}

.c1 option {
  font-weight: normal;
  cursor: pointer;
  text-align: center;
}

.c1 option:checked {
  font-weight: bold;
  cursor: pointer;
  background-color: white;
}

.c1 button {
  height: 40px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: white;
  border-radius: 8px;
  padding: 24px;
  box-shadow: 0px 0px 20px 0px lightgrey;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 img {
  max-width: 180px;
  max-height: 140px;
  margin: 16px;
}

.c4 {
  display: grid;
  width: 100%;
  gap: 16px;
  grid-template-columns: 1fr;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 button {
  margin-top: 12px;
}

@media (min-width:768px) {
  .c4 {
    grid-template-columns: 1fr 1fr 1fr;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <input
            placeholder="Nombre"
            value=""
          />
          <input
            placeholder="Razon social"
            value=""
          />
          <input
            placeholder="RUT"
            value=""
          />
          <select>
            <option
              disabled=""
              value=""
            >
              Score
            </option>
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
          </select>
          <select>
            <option
              disabled=""
              value=""
            >
              Rubro
            </option>
            <option
              value="Cat1"
            >
              Cat1
            </option>
          </select>
          <div
            style="gap: 8px; display: flex;"
          >
            <button
              action="secondary"
              class="c2"
              type="submit"
            >
              Limpiar
            </button>
            <button
              action="primary"
              class="c3"
              disabled=""
              type="submit"
            >
              Buscar
            </button>
          </div>
        </div>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <h3>
              Canal 10
            </h3>
            <img
              alt="Canal 10 Logo"
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/c8/Canal_10_Uruguay_Logo_1960.webp/640px-Canal_10_Uruguay_Logo_1960.webp.png"
            />
            <div
              class="c6"
            >
              <span>
                Score: 
                9
              </span>
              <span>
                Razon Social: 
                CANAL 10 S.A.
              </span>
              <span>
                RUT: 
                734879753738964
              </span>
              <button
                action="primary"
                class="c3"
                disabled=""
              >
                Ver perfil de proveedor
              </button>
            </div>
          </div>
          <div
            class="c5"
          >
            <h3>
              Canal 4
            </h3>
            <img
              alt="Canal 4 Logo"
              src="https://upload.wikimedia.org/wikipedia/commons/3/33/Canal4_uy.png"
            />
            <div
              class="c6"
            >
              <span>
                Score: 
                6
              </span>
              <span>
                Razon Social: 
                CANAL 4 S.A.
              </span>
              <span>
                RUT: 
                633334234363643
              </span>
              <button
                action="primary"
                class="c3"
                disabled=""
              >
                Ver perfil de proveedor
              </button>
            </div>
          </div>
          <div
            class="c5"
          >
            <h3>
              Teledoce
            </h3>
            <img
              alt="Teledoce Logo"
              src="https://upload.wikimedia.org/wikipedia/commons/7/79/Teledoce_2022.png"
            />
            <div
              class="c6"
            >
              <span>
                Score: 
                7
              </span>
              <span>
                Razon Social: 
                TELEDOCE S.A.
              </span>
              <span>
                RUT: 
                3457385937529
              </span>
              <button
                action="primary"
                class="c3"
                disabled=""
              >
                Ver perfil de proveedor
              </button>
            </div>
          </div>
          <div
            class="c5"
          >
            <h3>
              Banco Santander del Uruguay
            </h3>
            <img
              alt="Banco Santander del Uruguay Logo"
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/6/60/Banco_santander_logo.svg/2560px-Banco_santander_logo.svg.png"
            />
            <div
              class="c6"
            >
              <span>
                Score: 
                9
              </span>
              <span>
                Razon Social: 
                SANTANDER S.A.
              </span>
              <span>
                RUT: 
                5765384337675
              </span>
              <button
                action="primary"
                class="c3"
                disabled=""
              >
                Ver perfil de proveedor
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  gap: 64px;
  width: 100%;
}

.c2 {
  border-radius: 8px;
  height: 48px;
  font-size: 16px;
  padding: 12px 24px;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  border: 1.5px solid #04b0f4;
  background-color: white;
  color: #04b0f4;
}

.c2:disabled {
  border-color: #c0bfcb;
  color: #c0bfcb;
}

.c2:hover:enabled {
  cursor: pointer;
  color: white;
  background-color: #04b0f4;
}

.c2:active:enabled {
  color: white;
  background-color: #137ca5;
  border-color: #137ca5;
}

.c3 {
  border-radius: 8px;
  height: 48px;
  font-size: 16px;
  padding: 12px 24px;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  border: 1.5px solid #fca408;
  background-color: white;
  color: #fca408;
}

.c3:disabled {
  border-color: #c0bfcb;
  color: #c0bfcb;
}

.c3:hover:enabled {
  cursor: pointer;
  color: white;
  background-color: #fca408;
}

.c3:active:enabled {
  color: white;
  background-color: #b27508;
  border-color: #b27508;
}

.c1 {
  background: white;
  border-radius: 8px;
  padding: 24px;
  box-shadow: 0px 0px 20px 0px lightgrey;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 input {
  height: 40px;
  border-radius: 8px;
  border: 1px solid lightgray;
  padding-left: 12px;
  padding-right: 12px;
}

.c1 select {
  height: 40px;
  border-radius: 8px;
  border: 1px solid lightgray;
  padding-left: 12px;
  padding-right: 12px;
  cursor: pointer;
}

.c1 option {
  font-weight: normal;
  cursor: pointer;
  text-align: center;
}

.c1 option:checked {
  font-weight: bold;
  cursor: pointer;
  background-color: white;
}

.c1 button {
  height: 40px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: white;
  border-radius: 8px;
  padding: 24px;
  box-shadow: 0px 0px 20px 0px lightgrey;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 img {
  max-width: 180px;
  max-height: 140px;
  margin: 16px;
}

.c4 {
  display: grid;
  width: 100%;
  gap: 16px;
  grid-template-columns: 1fr;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 button {
  margin-top: 12px;
}

@media (min-width:768px) {
  .c4 {
    grid-template-columns: 1fr 1fr 1fr;
  }
}

<div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <input
          placeholder="Nombre"
          value=""
        />
        <input
          placeholder="Razon social"
          value=""
        />
        <input
          placeholder="RUT"
          value=""
        />
        <select>
          <option
            disabled=""
            value=""
          >
            Score
          </option>
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4"
          >
            4
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="6"
          >
            6
          </option>
          <option
            value="7"
          >
            7
          </option>
          <option
            value="8"
          >
            8
          </option>
          <option
            value="9"
          >
            9
          </option>
          <option
            value="10"
          >
            10
          </option>
        </select>
        <select>
          <option
            disabled=""
            value=""
          >
            Rubro
          </option>
          <option
            value="Cat1"
          >
            Cat1
          </option>
        </select>
        <div
          style="gap: 8px; display: flex;"
        >
          <button
            action="secondary"
            class="c2"
            type="submit"
          >
            Limpiar
          </button>
          <button
            action="primary"
            class="c3"
            disabled=""
            type="submit"
          >
            Buscar
          </button>
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <h3>
            Canal 10
          </h3>
          <img
            alt="Canal 10 Logo"
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/c8/Canal_10_Uruguay_Logo_1960.webp/640px-Canal_10_Uruguay_Logo_1960.webp.png"
          />
          <div
            class="c6"
          >
            <span>
              Score: 
              9
            </span>
            <span>
              Razon Social: 
              CANAL 10 S.A.
            </span>
            <span>
              RUT: 
              734879753738964
            </span>
            <button
              action="primary"
              class="c3"
              disabled=""
            >
              Ver perfil de proveedor
            </button>
          </div>
        </div>
        <div
          class="c5"
        >
          <h3>
            Canal 4
          </h3>
          <img
            alt="Canal 4 Logo"
            src="https://upload.wikimedia.org/wikipedia/commons/3/33/Canal4_uy.png"
          />
          <div
            class="c6"
          >
            <span>
              Score: 
              6
            </span>
            <span>
              Razon Social: 
              CANAL 4 S.A.
            </span>
            <span>
              RUT: 
              633334234363643
            </span>
            <button
              action="primary"
              class="c3"
              disabled=""
            >
              Ver perfil de proveedor
            </button>
          </div>
        </div>
        <div
          class="c5"
        >
          <h3>
            Teledoce
          </h3>
          <img
            alt="Teledoce Logo"
            src="https://upload.wikimedia.org/wikipedia/commons/7/79/Teledoce_2022.png"
          />
          <div
            class="c6"
          >
            <span>
              Score: 
              7
            </span>
            <span>
              Razon Social: 
              TELEDOCE S.A.
            </span>
            <span>
              RUT: 
              3457385937529
            </span>
            <button
              action="primary"
              class="c3"
              disabled=""
            >
              Ver perfil de proveedor
            </button>
          </div>
        </div>
        <div
          class="c5"
        >
          <h3>
            Banco Santander del Uruguay
          </h3>
          <img
            alt="Banco Santander del Uruguay Logo"
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/6/60/Banco_santander_logo.svg/2560px-Banco_santander_logo.svg.png"
          />
          <div
            class="c6"
          >
            <span>
              Score: 
              9
            </span>
            <span>
              Razon Social: 
              SANTANDER S.A.
            </span>
            <span>
              RUT: 
              5765384337675
            </span>
            <button
              action="primary"
              class="c3"
              disabled=""
            >
              Ver perfil de proveedor
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
